src/index.ts

```diff
@@ -40,6 +40,57 @@ function bundle(
     );
 }
 
+interface CompileApiInterfaceResult {
```

Comment:
If you have a function that returns this interface, you should export the interface as well (and also ExegesisCompiledOptions and PluginsManager).  Or, alternatively, you maybe want to not export this.  :P  Do you need all three of these for what you're doing, or just the apiInterface?

Reply:
I'd rather not expose them if I don't have to.  Less surface area for breaking changes.  :)

Reply:
I made it so the new exported function only returns the `apiInterface`. Then I realized I need to export a bunch of other types (involving `ApiInterface`), so you don't have a `types/internal.ts` file anymore 😛